### PR TITLE
python3+django2 support

### PR DESCRIPTION
**Pull request by thraxil**
* ran 2to3
* expanded tox/travis tests to run against python2 and 3

### CODE REVIEW

File: launchcontainer/launchcontainer.py [hunk 1]
```diff
@@ -64,6 +68,8 @@ def get_api_root_url(url):
 
 def is_valid(url):
     """Return True if this URL is valid."""
+    if url is None:
```
Comment (OmarIthawi) [inline launchcontainer/launchcontainer.py]:
```suggestion
    if not url:
```

File: launchcontainer/tests.py [hunk 2]
```diff
@@ -55,7 +55,7 @@ def make_one(self, display_name=None, **kw):
         """
         Creates a launchcontainer XBlock for testing purpose.
         """
-        from launchcontainer import LaunchContainerXBlock as cls
+        from .launchcontainer import LaunchContainerXBlock as cls
```
Comment (OmarIthawi) [inline launchcontainer/tests.py:58]: This is weird. I think `launchcontainer` is available without the dot prefix and this is just 2to3 noise. Correct?

```suggestion
        from launchcontainer import LaunchContainerXBlock as cls
```

File: tox.ini [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
-envlist = py27,flake8
+envlist = py27-django111,py3-django2,flake8
 skipsdist = True
 
 [testenv]
-deps = -r{toxinidir}/requirements/test.txt
-       Django<2
+deps =
+    django111: Django<2
+    django22: Django<3
```
Comment (OmarIthawi) [inline tox.ini:8]: More explicit requirements.

```suggestion
    django111: Django>=1.11,<2.0
    django22: Django>=2.0,<2.3
```

File: .travis.yml [hunk 4]
```diff
@@ -5,12 +5,10 @@ python:
...
   - pip install tox
 
-script:
-  - python run_tests.py
-
 env:
   - TOXENV=flake8
-  - TOXENV=py27
+  - TOXENV=py27-django111
+  - TOXENV=py3-django2
```
Comment (OmarIthawi) [inline .travis.yml:11]: So `py3` would run _any_ Python 3 available, right? For this XBlock I think it's good.

**Review by OmarIthawi — APPROVED** [review]
Thanks @thraxil! This is great!

I've added mostly just minor comments, so this should be good to merge as-is.

**Review by tkeemon — APPROVED** [review]

**Comment by briandant** [comment]
Go for it!

briandant.com <http://briandant.com>
778.681.7864

On November 16, 2020, GitHub Notifications <notifications@github.com>
wrote:
> @tkeemon approved this pull request.
>

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub
<https://github.com/appsembler/xblock-
launchcontainer/pull/45#pullrequestreview-532501351>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AAOMFZSD3QTQSU4YA3HCF4LSQKKMVANCNFSM4TXHGM5A>.

**Pull request merged**